src/api/users.rs

```diff
@@ -56,6 +57,51 @@ pub async fn delete_user(
     Ok(HttpResponse::Ok().finish())
 }
 
+#[derive(Serialize)]
+pub struct CurrentHeartBeat {
+    pub started: chrono::NaiveDateTime,
+    pub duration: i64,
+    pub heartbeat: HeartBeat,
+}
+
+#[get("/users/{username}/activity/current")]
+pub async fn get_current_activity(
+    path: Path<(String,)>,
+    user: UserId,
+    db: Data<DbPool>,
+    heartbeats: Data<HeartBeatMemoryStore>,
+) -> Result<impl Responder, TimeError> {
+    let mut conn = db.get()?;
+
+    let target_user = if path.0 == "@me" {
+        user.id
+    } else {
+        let target_user = conn.get_user_by_name(&path.0)?;
+        if target_user.id == user.id
+            || target_user.is_public
+            || block(move || conn.are_friends(user.id, target_user.id)).await??
+        {
+            target_user.id
+        } else {
+            return Err(TimeError::Unauthorized);
+        }
+    };
+
+    match heartbeats.get(&target_user) {
+        Some(a) => {
```

Comment:
Please fix the variable names, what the hell does `a` or `rs` mean

`started` is confusing too

Reply:
Perhaps `heartbeat`, `last_heartbeat`, and `session_start` instead of `a`, `heartbeat`, and `started`, respectively?

Reply:
Seems good @lajp godo